packages/common/test/util/http.test.js

```diff
@@ -0,0 +1,449 @@
+
+      expect(result.data).to.eql('hello world');
+    });
+
+    // parameters needs to set parseAs
+    it('should force as json', async () => {
+      client
+        .intercept({
+          path: '/api',
+          method: 'GET',
+        })
+        .reply(200, {
+          name: 'aissa',
+        });
+
+      const result = await request('GET', 'https://www.example.com/api', {
+        parseAs: 'json',
+      });
+
+      expect(result.data).to.eql({ name: 'aissa' });
+    });
+    // explicitly include headers to ensure they're ignore?
+    it('should force as json even if content type is string', async () => {
+      client
+        .intercept({
+          path: '/api',
+          method: 'GET',
+        })
+        .reply(
+          200,
+          {
+            name: 'aissa',
+          },
+          {
+            headers: {
+              'content-type': 'application/text',
+            },
+          }
+        );
+
+      const result = await request('GET', 'https://www.example.com/api', {
+        parseAs: 'json',
+      });
+
+      expect(result.data).to.eql({ name: 'aissa' });
+    });
+    it('should force as stream', async () => {
+      client
+        .intercept({
+          path: '/api',
+          method: 'GET',
+        })
+        .reply(200, {
+          name: 'iam stream',
+        });
+
+      const result = await request('GET', 'https://www.example.com/api', {
+        parseAs: 'stream',
+      });
+
+      //TODO - Better technique for testing stream
+      expect(await result.data.json()).to.eql({ name: 'iam stream' });
+    });
+  });
+
+  it('should make a successful arbitary request', async () => {
+    client
+      .intercept({
+        path: '/api',
+        method: 'JOE',
+      })
+      .reply(
+        200,
+        {},
+        {
+          headers: {
+            'content-type': 'application/json',
+          },
+        }
+      );
+
+    const result = await request('JOE', 'https://www.example.com/api');
+
+    expect(result.data).to.eql({});
+    expect(result.code).to.eql(200);
+  });
+
+  it('should use baseUrl from options', async () => {
+    client
+      .intercept({
+        path: '/api',
+        method: 'GET',
+      })
+      .reply(200, {});
+
+    const response = await request('GET', '/api', {
+      baseUrl: 'https://www.example.com',
+    });
+
+    expect(response.code).to.eql(200);
+  });
+
+  it('should accept an absolute url', async () => {
+    client
+      .intercept({
+        path: '/api',
+        method: 'GET',
+      })
+      .reply(200, {});
+
+    const response = await request('GET', 'https://www.example.com/api');
+
+    expect(response.code).to.eql(200);
+  });
+});
+
+describe('options', () => {
+  it('should not throw 204 if response is okay using errorMap', async () => {
+    client
+      .intercept({
+        path: '/api/content',
+        method: 'GET',
+      })
+      .reply(
+        200,
+        {},
```

Comment:
I don't think any of these error map tests need response data or headers? Can we check and remove if possible?

Test code should be as small and accurate as possible!

Reply:
Removed the headers and the test are still 👍🏽